### DISCUSSION TITLE
CoreWCF in Windows Service using the BackgroundService implementation.

### DISCUSSION THREAD

**Discussion by demeterzsolt** [CoreWCF/CoreWCF#1406] [Q&A] [answered]
Hello Guys,

Can you provide a complete example about CoreWCF configuration and implementation running as Windows Service (using the BackgroundService pattern), please ?

**Comment by ShadyHarbinger** [comment]
I am interested in this one as well. SO many people asked for still no answer.

**Reply by g7ed6e** [reply, marked as answer]
You can find a sample in this PR in the samples repo. https://github.com/CoreWCF/samples/pull/26

**Reply by demeterzsolt** [reply]
Thanks !

**Reply by ShadyHarbinger** [reply]
> You can find a sample in this PR in the samples repo. [CoreWCF/samples#26](https://github.com/CoreWCF/samples/pull/26)

It is for Framework not .NET. Is there any CoreWCF .NET 6-8 sample as Windows Service?

**Edit:** Ok. I was blind. The solution is was simple. I do not know why I did not think of it:
```
var builder = WebApplication.CreateBuilder(args);
builder.Services.AddWindowsService(options => options.ServiceName = "WcfService");
```

**Reply by AlleSchonWeg** [reply]
Hi @g7ed6e 
thank you for the sample. I have a question. Could you tell me what the benefit is to use a BackgroundService with CoreWCF if i want to host the app as a windows service? I mean the WindowsServiceWorker only output log messages in a infinity loop until the service is shut down. 
Thank you
